torchtune/modules/attention.py

```diff
@@ -255,6 +255,14 @@ def forward(
                 )
             k = self.kv_cache.k_cache
             v = self.kv_cache.v_cache
+
+            # If needed, expand the key and value tensors to have the same shape
+            # as the query tensor by copying values across the relevant dim
+            # k,v shape: [b, n_kv, s, h_d] -> [b, n_h, s, h_d]
+            if self.num_heads != self.num_kv_heads:
```

Comment:
Is there a test that can catch the error you noticed before you made this change?

Reply:
The tests in this PR will catch it (it's how I found it) since we perform generation with a vision model with KV-caches enabled. The most atomic test that *should* catch it is setting up a cross attention layer with KV-caches and kv_heads < num_heads somewhere like [here](https://github.com/pytorch/torchtune/blob/main/tests/torchtune/modules/test_transformer_decoder.py)